geomstats/geometry/complex_poincare_disk.py

```diff
@@ -0,0 +1,360 @@
+    space of dimension 2. Its complex dimension is 1.
+
+    Parameters
+    ----------
+    scale : float
+        Scale of the complex Poincaré metric.
+        Optional, default: 1.
+    """
+
+    def __init__(self, scale=1.0, **kwargs):
+        super().__init__(
+            dim=1,
+            embedding_space=Hermitian(dim=1),
+            metric=ComplexPoincareDiskMetric(scale=scale),
+            **kwargs
+        )
+        self.scale = scale
+
+    @staticmethod
+    def belongs(point, atol=gs.atol):
+        """Check if a point belongs to the complex unit disk.
+
+        Check if a point belongs to the Poincaré complex disk,
+        i.e. Check if its norm is lower than one.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point to be checked.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean denoting if point belongs to the
+            complex Poincaré disk.
+        """
+        is_scalar = gs.ndim(point) == 1
+        is_scalar = is_scalar or (gs.ndim(point) == 2 and point.shape[1] == 1)
+        has_modulus_lower_than_one = gs.array(
+            [[gs.all(gs.abs(point[i_point]) < 1)] for i_point in range(point.shape[0])]
+        )
+        belongs = gs.logical_and(is_scalar, has_modulus_lower_than_one)
+        belongs = gs.reshape(belongs, (-1,))
+        return belongs
+
+    @staticmethod
+    def projection(point):
+        """Project a point on the complex unit disk.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point to project.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        projected : array-like, shape=[..., 1]
+            Projected point.
+        """
+        return gs.where(
+            gs.abs(point) >= 1 - gs.atol,
+            gs.cast((1 - gs.atol) / gs.abs(point), dtype=point.dtype) * point,
+            point,
+        )
+
+    @staticmethod
+    def random_point(n_samples=1, bound=0.8):
+        """Generate random points in the complex unit disk.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples.
+            Optional, default: 1.
+        bound : float
+            Bound of the interval in which to sample in the tangent space.
+            Optional, default: 0.8.
+
+        Returns
+        -------
+        samples : array-like, shape=[..., 1]
+            Points sampled in the unit disk.
+        """
+        size = (n_samples, 1) if n_samples != 1 else (1,)
+
+        modulus = gs.random.rand(*size, dtype=gs.get_default_cdtype())
+        modulus *= bound
+        angle = 2 * gs.pi * gs.random.rand(*size, dtype=gs.get_default_cdtype())
+        samples = modulus * gs.exp(1j * angle)
+        return samples
```

Comment:
I find the shape of this return unexpected, I would expect it to be `[...]` as a point is not an array, so batch is 1d array.

Reply:
Done.